internal/integration/spec_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2019 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package integration
+
+import (
+	"testing"
+
+	"github.com/newrelic/nri-prometheus/internal/pkg/labels"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLoadSpecFilesOk(t *testing.T) {
+	specs, err := LoadSpecFiles("./test/")
+	assert.NoError(t, err)
+	assert.Contains(t, specs.SpecsByName, "ibmmq")
+	assert.Contains(t, specs.SpecsByName, "ravendb")
+	assert.Contains(t, specs.SpecsByName, "redis")
+}
+func TestLoadSpecFilesNoFiles(t *testing.T) {
+	specs, err := LoadSpecFiles(".")
+	assert.NoError(t, err)
+	assert.Len(t, specs.SpecsByName, 0)
+}
+
+func TestSpecs_getEntity(t *testing.T) {
+
+	specs, err := LoadSpecFiles("./test/")
+	assert.NoError(t, err)
+
+	type fields struct {
+		SpecsByName map[string]SpecDef
+	}
+	type args struct {
+		m Metric
+	}
+	tests := []struct {
+		name           string
+		fields         fields
+		args           args
+		wantEntityName string
+		wantEntityType string
+		wantErr        bool
+	}{
+		{
+			name:   "matchEntity",
+			fields: fields{specs.SpecsByName},
+			args: args{
+				Metric{
+					name: "ravendb_database_document_put_bytes_total",
+					attributes: labels.Set{
+						"database": "test",
+					},
+				}},
+			wantEntityName: "database:test",
+			wantEntityType: "PrometheusRavendbDatabase",
+			wantErr:        false,
+		},
+		{
+			name:   "matchEntityWithoutDimensions",
+			fields: fields{specs.SpecsByName},
+			args: args{
+				Metric{
+					name:       "ravendb_document_put_bytes_total",
+					attributes: labels.Set{},
+				}},
+			wantEntityName: "node",
+			wantEntityType: "PrometheusRavendbNode",
```

Comment:
Are we going to remove `Prometheus` prefix from entity name in next iteration?

Reply:
yes i'll do it